framework/src/interfaces/TaggingInterface.C

```diff
@@ -127,9 +130,12 @@ TaggingInterface::TaggingInterface(const MooseObject * moose_object)
 
   const auto * const fe_problem =
       moose_object->parameters().getCheckedPointerParam<FEProblemBase *>("_fe_problem_base");
-  if (const auto * const ref_problem = dynamic_cast<const ReferenceResidualProblem *>(fe_problem))
+
+  if (fe_problem->hasConvergence("reference_residual"))
```

Comment:
We shouldn't rely on anyone using a particular name for the convergence object. And what we need to do here is loop over ALL convergence objects, and check if each is of type `ReferenceResidualConvergence` and do these things if so.